coreos-meta-translator/trans.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+from argparse import ArgumentParser, FileType
+from datetime import datetime
+import json
+import os
+
+
+def ensureDup(inp, out, inp_key, out_key):
+    if out.get(out_key, None) == None:
+        out[out_key] = inp.get(inp_key)
+        return out
+    if out.get(out_key) != inp.get(inp_key):
+        print("Input Files do not appear to be for the same release")
+        raise SystemExit(1)
+    return out
+
+def url_builder(stream, version, arch, path):
+    return f"https://fcos-builds.s3.amazonaws.com/prod/streams/{stream}/builds/{version}/{arch}/{path}"
+
+def get_extension(path, ext_count):
+    return ".".join(path.split(".")[-ext_count:])
+
+
+parser = ArgumentParser()
+parser.add_argument("--workdir", help="cosa workdir", required=True)
+parser.add_argument("--build-id", help="build id", required=True)
+args = parser.parse_args()
+
+outer_dir = os.path.join(args.workdir, "builds", args.build_id)
+release_file = os.path.join(outer_dir, "release.json")
+
+out = {}
+if os.path.exists(release_file):
+    with open(release_file, 'r') as w:
+        out = json.load(w)
+
+build_dirs = [os.path.join(outer_dir, arch_dir) for arch_dir in os.listdir(outer_dir) if os.path.isdir(os.path.join(outer_dir, arch_dir))]
+files = [os.path.join(build_dir, "meta.json") for build_dir in build_dirs if os.path.exists(os.path.join(build_dir, "meta.json"))]
```

Comment:
Another (somewhat more correct) approach is to get the list of architectures for which the build exists by checking `archs`.

Reply:
Where does the `archs` key exist?

Reply:
Check out https://github.com/coreos/coreos-assembler/pull/580#issue-292980446.